Chat_App/Chat_App/UI/MessageTableViewCell.swift

```diff
@@ -0,0 +1,310 @@
+        setUpLeftSmallBubbleConstraints()
+        setUpLeftMiddleBubbleConstraints()
+        setUpLeftChatBubbleConstraints()
+        setUpLeftMessageLabelConstraints()
+        setUpLeftDateLabelConstraints()
+    }
+
+    private func setUpLeftSmallBubbleConstraints() {
+        NSLayoutConstraint.activate([
+            smallBubble.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: SmallBubbleConstants.leadingAnchor),
+            smallBubble.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: SmallBubbleConstants.bottomAnchor),
+            smallBubble.widthAnchor.constraint(equalToConstant: SmallBubbleConstants.widthHeightAnchor),
+            smallBubble.heightAnchor.constraint(equalToConstant: SmallBubbleConstants.widthHeightAnchor)
+        ])
+    }
+
+    private func setUpLeftMiddleBubbleConstraints() {
+        NSLayoutConstraint.activate([
+            middleBubble.leadingAnchor.constraint(equalTo: chatBubble.leadingAnchor, constant: MiddleBubbleConstants.leadingAnchor),
+            middleBubble.bottomAnchor.constraint(equalTo: chatBubble.bottomAnchor, constant: MiddleBubbleConstants.bottomAnchor),
+            middleBubble.widthAnchor.constraint(equalToConstant: MiddleBubbleConstants.widthHeightAnchor),
+            middleBubble.heightAnchor.constraint(equalToConstant: MiddleBubbleConstants.widthHeightAnchor)
+        ])
+    }
+
+    private func setUpLeftChatBubbleConstraints() {
+        NSLayoutConstraint.activate([
+            chatBubble.topAnchor.constraint(equalTo: contentView.topAnchor, constant: ChatBubbleConstants.topAnchor),
+            chatBubble.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: ChatBubbleConstants.leadingAnchor),
+        ])
+    }
+
+    private func setUpLeftMessageLabelConstraints() {
+        NSLayoutConstraint.activate([
+            messageLabel.topAnchor.constraint(equalTo: chatBubble.topAnchor, constant: MessageLabelConstants.topAnchor),
+            messageLabel.leadingAnchor.constraint(equalTo: chatBubble.leadingAnchor, constant: MessageLabelConstants.leadingAnchor),
+            messageLabel.trailingAnchor.constraint(equalTo: chatBubble.trailingAnchor, constant: MessageLabelConstants.trailingAnchor),
+            messageLabel.bottomAnchor.constraint(equalTo: chatBubble.bottomAnchor, constant: MessageLabelConstants.bottomAnchor),
+        ])
+    }
+
+    private func setUpLeftDateLabelConstraints() {
+        NSLayoutConstraint.activate([
+            dateLabel.topAnchor.constraint(equalTo: chatBubble.bottomAnchor, constant: DateLabelConstants.topAnchor),
+            dateLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: DateLabelConstants.leadingAnchor),
+            dateLabel.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: DateLabelConstants.bottomAnchor),
+        ])
+    }
+}
+
+// MARK: Constants
+private extension MessageTableViewCell {
+
```

Comment:
```suggestion
enum Constants {
    enum ChatBubble {
    // ...
    }
    
    // and so on
}
```
Just a suggestion to have a common `Constants` namespace